File: app/src/test/java/com/mower/MowerAppShould.java
package com.mower;

import com.mower.application.ConsoleApplication;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;

import static org.mockito.Mockito.doNothing;
import static org.mockito.Mockito.verify;

@ExtendWith(MockitoExtension.class)
class MowerAppShould {
  @Mock
  ConsoleApplication consoleApplicationMocked;

  @Test
  void main() {
    doNothing().when(consoleApplicationMocked).start();
    MowerApp.initForTestPurposes(consoleApplicationMocked);
    MowerApp.main();
    verify(consoleApplicationMocked).start();
  }
}

File: app/src/test/kotlin/com/mower/MowerAppShould.kt
package com.mower

import com.mower.application.ConsoleApplication
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.mockito.Mockito
import org.mockito.Mockito.doNothing
import org.mockito.Mockito.mock
import org.mockito.junit.jupiter.MockitoExtension

@ExtendWith(MockitoExtension::class)
class MowerAppShould {
    private val consoleApplicationMocked: ConsoleApplication = mock(ConsoleApplication::class.java)

    @Test
    fun main() {
        doNothing().`when`(consoleApplicationMocked).start()
        val mower = MowerApp()
        mower.initForTestPurposes(consoleApplicationMocked)
        mower.main()
        Mockito.verify(consoleApplicationMocked).start()
    }
}
